com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/tile/TileSetGenerator.java

```diff
@@ -161,8 +161,7 @@ private static int calcTileStart(TileSet tileSet, int size, int tileIndex) {
         int tileHeight = tileSet.getTileHeight();
         List<BufferedImage> result = new ArrayList<BufferedImage>(count);
         for (int i = 0; i < count; ++i) {
-            int type = TextureUtil.getImageType(image);
-            BufferedImage tgt = new BufferedImage(tileWidth, tileHeight, type);
+            BufferedImage tgt = new BufferedImage(tileWidth, tileHeight, BufferedImage.TYPE_4BYTE_ABGR);
```

Comment:
For indexed (palette) image the problem was this bit of code where we created another indexed image and copied the subimage. The new subimage always seems to result in an image with a black background. I tried setting a transparent background color and clearing the image before copying,  but it didn't help.

Changing to BufferedImage.TYPE_4BYTE_ABGR works and in the end it is the output format for the final texture so we might as well convert it here.

Reply:
Is this the only place where we needed to fix it?

> in the end it is the output format for the final texture so we might as well convert it here.

Not sure how to interpret that?
So, even if I input an RGB texture, I'll always get a RGBA texture out?

Reply:
> So, even if I input an RGB texture, I'll always get a RGBA texture out?

Yes, the tilesource image will be split into multiple images, then passed to `TextureSetGenerator.generate()` which will call the `composite()` function. The `composite()` function always creates a `BufferedImage` of type `TYPE_4BYTE_ABGR`:

https://github.com/defold/defold/blob/dev/com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/textureset/TextureSetGenerator.java#L478-L487

Reply:
> Is this the only place where we needed to fix it?

Yes, as far as I can tell. For normal atlas creation we do not have this problem (see above).